src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,856 @@
+				}
+			}
+
+			var childIndexInFlatTree = GetNextIndexInFlatTree(node) + childIndex + allOpenedDescendantsCount;
+			return GetNodeAt(childIndexInFlatTree);
+		}
+
+		private int CountDescendants(TreeViewNode value)
+		{
+			int descendantCount = 0;
+			var size = value.Children.Count;
+			for (var i = 0; i < size; i++)
+			{
+				var childNode = value.Children[i];
+				descendantCount++;
+				if (childNode.IsExpanded)
+				{
+					descendantCount = descendantCount + CountDescendants(childNode);
+				}
+			}
+
+			return descendantCount;
+		}
+
+		private int IndexOfNextSibling(TreeViewNode childNode)
+		{
+			var parentNode = childNode.Parent;
+			int stopIndex;
+			bool isLastRelativeChild = true;
+			while (parentNode != null && isLastRelativeChild)
+			{
+				int relativeIndex = parentNode.Children.IndexOf(childNode);
+				if (parentNode.Children.Count - 1 != relativeIndex)
+				{
+					isLastRelativeChild = false;
+				}
+				else
+				{
+					childNode = parentNode;
+					parentNode = parentNode.Parent;
+				}
+			}
+
+			if (parentNode != null)
+			{
+				int siblingIndex = parentNode.Children.IndexOf(childNode);
+				var siblingNode = parentNode.Children[siblingIndex + 1];
+				IndexOfNode(siblingNode, out stopIndex);
+			}
+			else
+			{
+				stopIndex = Size;
+			}
+
+			return stopIndex;
+		}
+
+		private int GetExpandedDescendantCount(TreeViewNode parentNode)
+		{
+			var allOpenedDescendantsCount = 0;
+			for (var i = 0; i < parentNode.Children.Count; i++)
+			{
+				var childNode = (TreeViewNode)parentNode.Children[i];
+				allOpenedDescendantsCount++;
+				if (childNode.IsExpanded)
+				{
+					allOpenedDescendantsCount += CountDescendants(childNode);
+				}
+			}
+			return allOpenedDescendantsCount;
+		}
+
+		internal bool IsNodeSelected(TreeViewNode targetNode)
+		{
+			return m_selectedNodes.IndexOf(targetNode) != -1;
+		}
+
+		private TreeNodeSelectionState NodeSelectionState(TreeViewNode targetNode)
+		{
+			return targetNode.SelectionState;
+		}
+
+		private void UpdateNodeSelection(TreeViewNode selectNode, TreeNodeSelectionState selectionState)
+		{
+			var node = selectNode;
+			if (selectionState != node.SelectionState)
+			{
+				node.SelectionState = selectionState;
+				var selectedNodes = m_selectedNodes;
+				switch (selectionState)
+				{
+					case TreeNodeSelectionState.Selected:
+						selectedNodes.InsertAtCore(selectedNodes.Count, selectNode);
+						//TODO
+						//m_selectedNodeChildrenChangedEventTokenVector.push_back(get_self<TreeViewNode>(selectNode).ChildrenChanged({ this, &ViewModel.SelectedNodeChildrenChanged }));
+						break;
+
+					case TreeNodeSelectionState.PartialSelected:
+					case TreeNodeSelectionState.UnSelected:
+						int index = selectedNodes.IndexOf(selectNode);
+						if (index > -1)
+						{
+							selectedNodes.RemoveAtCore(index);
+							//TODO
+							//selectNode.ChildrenChanged(m_selectedNodeChildrenChangedEventTokenVector[index]);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)